### ISSUE TITLE
Can't get object-valued property from method call on UI5 control

### ISSUE THREAD

**Issue by ph-alsvik** [js-soft/wdi5#190]
**Describe the bug**
There is a (somewhat) rare case when the returned value of some UI5 Control method is an object, and not a JS primitive or another ui5 Control. For example ```getMaxDate``` on ```sap.m.DatePicker``` returns a JS Date object. In this case, wdi5 doesn’t pick up on this:

- In case when this property hasn’t been set, the returned value is null, and wdi5 logs it as an error.

from client-side-js/executeControlMethod.js:
```javascript
} else if (result === undefined || result === null) {
    done([
        "error",
        `function ${methodName} does not exist on control ${metadata.getElementName()}!`,
        "none"
    ])
```

- In case when it has been set, the returned value is assumed to be the control itself.

```javascript
} else {
    done([
        "success",
        `instance of wdi5 representation of a ${metadata.getElementName()}`,
        "element"
    ])
}
```

**To Reproduce**

I used [this sample app](https://sapui5.hana.ondemand.com/#/entity/sap.m.DatePicker/sample/sap.m.sample.DatePicker) from SAPUI5 SDK, and ran this test against it:

```javascript
describe("Date picker test", () => {
    it("should get the max date of the date picker", async () => {
        const oDatePickerSelector = {
            selector: {
                id: "DP1",
                controlType: "sap.m.DatePicker",
                viewName: "sap.m.sample.DatePicker.Group"
            }
        };
        const oDatePicker = await browser.asControl(oDatePickerSelector);
        const maxDate = await oDatePicker.getMaxDate();
    });
});
```

**Expected behavior**
Should return null if max date not set, or a JS Date object.

**Logs/Console Output**

```shell
[0-0] 2022-03-18T14:00:07.846Z INFO webdriver: RESULT [
[0-0]   'error',
[0-0]   'function getMaxDate does not exist on control sap.m.DatePicker!',
[0-0]   'none'
[0-0] ]
[0-0] [wdi5] call of getMaxDate failed because of: function getMaxDate does not exist on control sap.m.DatePicker!
[0-0] 2022-03-18T14:00:07.850Z INFO webdriver: COMMAND deleteSession()
[0-0] 2022-03-18T14:00:07.851Z INFO webdriver: [DELETE] http://localhost:9515/session/1223fa87118fa71cebbb51d7b0407ea5
```

**Comment by vobu** [comment]
Thanks once again for your efforts and detailed analysis of the bug: double-👍 
we'll tackle this.

**Comment by dominikfeininger** [comment]
Yep, we cant't return any other than primitive values from browser to node scope. Some alternatives might be either use eg. UNIX Timestamp and convert back and forth. Or try to stringify object and parse it back. I will do some investigation.

**Comment by vobu** [comment]
this is now enabled (returning non-primitives) and was released with `v0.9.0-rc4.2`
